### ISSUE TITLE
Error establishing database connection.

### ISSUE THREAD

**Issue by v3nt** [urre/wordpress-nginx-docker-compose#12]
Great tutorial! 

Everything works except the Wp install can't connect to the database.

https://myapp.local

Gives me

````
Warning: mysqli_real_connect(): php_network_getaddresses: getaddrinfo failed: Name or service not known in /var/www/html/web/wp/wp-includes/wp-db.php on line 1612 Warning: mysqli_real_connect(): (HY000/2002): php_network_getaddresses: getaddrinfo failed: Name or service not known in /var/www/html/web/wp/wp-includes/wp-db.php on line 1612

php_network_getaddresses: getaddrinfo failed: Name or service not known

````

or

````
Warning: mysqli_real_connect(): (HY000/2002): Connection refused in /var/www/html/web/wp/wp-includes/wp-db.php on line 1612
Connection refused
````

```
Error establishing a database connection
This either means that the username and password information in your wp-config.php file is incorrect or we can’t contact the database server at db:3306. This could mean your host’s database server is down.

Are you sure you have the correct username and password?
Are you sure that you have typed the correct hostname?
Are you sure that the database server is running?
```

All DB credentials are the same. Changed everything to 'wordpress' but the wp install still wont connect. Is there a way to see what the containers are setting the login info as? Or see the generated wp-config.php settings?

I tried adding a phpmyadmin container. It was visible but couldn't log in.

This is by far the closest tutorial for what i need & would be amazing to get over this final hurdle. I've been using docker a lot and not had this issue before. However none of my previous projects use a local domain or ssl.

D.

docker-compose.yml

````
version: '3.6'
services:
  nginx:
    image: nginx:latest
    container_name: myapp-nginx
    ports:
      - '8084:80'
      - '443:443'
    volumes:
      - ./nginx:/etc/nginx/conf.d
      - ./src:/var/www/html:rw,cached
      - ./certs:/etc/certs
    depends_on:
      - wordpress
    restart: always

  db:
    image: mysql:5.7
    container_name: myapp-mysql
    volumes:
       - './data/db:/var/lib/mysql:delegated'
    environment:
      - MYSQL_ROOT_PASSWORD=wordpress
      - MYSQL_DATABASE=wordpress
      - MYSQL_USER=wordpress
      - MYSQL_PASSWORD=wordpress
    restart: always
    # ports:
    #   - '3306:3306'

  wordpress:
    build:
      context: .
      dockerfile: Dockerfile
    container_name: myapp-wordpress
    volumes:
      - ./src:/var/www/html:rw,cached
      - ./config/php.ini:/usr/local/etc/php/conf.d/php.ini
    environment:
      - WORDPRESS_DB_NAME=wordpress
      - WORDPRESS_TABLE_PREFIX=wp_
      - WORDPRESS_DB_HOST=db:3306
      - WORDPRESS_DB_PASSWORD=wordpress
    depends_on:
      - db
    restart: always

  composer:
    image: composer/composer
    container_name: myapp-composer
    working_dir: /var/www/html
    restart: 'no'
    volumes:
      - ./src:/var/www/html:rw,cached
````

.env

````
DB_HOST=db:3306
DB_NAME=wordpress
DB_USER=wordpress
DB_PASSWORD=wordpress
# DB_PREFIX=wp_

WP_ENV=development
WP_HOME=https://myapp.local
WP_SITEURL=${WP_HOME}/wp

# Generate your keys here: https://roots.io/salts.html
AUTH_KEY='generateme'
SECURE_AUTH_KEY='generateme'
LOGGED_IN_KEY='generateme'
NONCE_KEY='generateme'
AUTH_SALT='generateme'
SECURE_AUTH_SALT='generateme'
LOGGED_IN_SALT='generateme'
NONCE_SALT='generateme'

````

feedback during build

````
docker-compose up -d --force-recreate --build
Creating network "wordpress-nginx-docker-compose_default" with the default driver
Building wordpress
Step 1/4 : FROM urre/wordpress-nginx-docker-compose-image
 ---> c3797de1da74
Step 2/4 : RUN curl -o /bin/wp-cli.phar https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli.phar
 ---> Using cache
 ---> 2c780d637b20
Step 3/4 : RUN chmod +x /bin/wp-cli.phar
 ---> Using cache
 ---> 632824aec449
Step 4/4 : RUN cd /bin && mv wp-cli.phar wp
 ---> Using cache
 ---> 7a7b5fb81401

Successfully built 7a7b5fb81401
Successfully tagged wordpress-nginx-docker-compose_wordpress:latest
Creating myapp-composer  ... done
Creating myapp-mysql    ... done
Creating myapp-wordpress ... done
Creating myapp-nginx     ... done
````

**Comment by urre** [comment]
@v3nt hey! I'm happy to help.

The port number must match. You've commented that out in your code example above. 
So in your ENV file in `./src/.env` it should be `DB_HOST=mysql:3306`. Also in `docker-compose.yml` the port number should be `3306`

Like this:

```
ports:
      - '3306:3306'
```

**Comment by v3nt** [comment]
thanks - I did that to see if it made a difference and it doesn't. I also pulled in some working DB / MYSQL setup code from other project but I have the same issue.

**Comment by urre** [comment]
@v3nt It should work fine. Must be something in your setup. You've renamed your service from `mysql` to `db`. Did you reflect that in your `.env` file?

**Comment by urre** [comment]
@v3nt it should work fine. Did you remove your comments for the ports in `docker-compose.yml`?
The port must be set and the same. Double check so you're not using that port for something else.

**Comment by v3nt** [comment]
Yes i started completely from scratch using the repo. Same issue. Anyway I can debug the ports being used issue or chan ge them to something else?

On another note just tried on my old laptop and works so isolated issue on my main machine.

**Comment by urre** [comment]
Ok, haven't upgraded to Catalina yet.

Can you try this:
1.
Stop all containers using `docker container stop $(docker container ls -aq)`
Do `docker-compose up` again (or `docker-compose up -d`)
Then check if anything else is running on 3306 on your mac with `sudo lsof -i tcp:3306`. It should show your docker database service
Do `docker-compose ps` to verify

Another question: can you verify this works: https://docs.docker.com/compose/wordpress/
(a minimal wp setup)

**Comment by v3nt** [comment]
thanks Urre.

I've started from scratch but then nginx wouldn't start as i guess the port is in use even when no containers up.

Only `docker-compose down` seems to stop the containers in that directory.

I've changed nginx port to 8080:80 (a guess) and we're up and running.

Then ran `docker-compose up -d --force-recreate --build`

https://myappwonkhedev.local/  now works!

Eventually i will have most of my project running from docker like this but i guess the ports will always be an issue?

thanks for your patience!
